### PR TITLE
feat: add save-peer option to add subcommand

### PR DESCRIPTION
**Pull request by vramana**

### CODE REVIEW

File: crates/cli/src/commands/add.rs [hunk 1]
```diff
@@ -23,6 +23,9 @@ pub struct AddCommandArgs {
     /// Install the specified packages as optionalDependencies.
     #[arg(short = 'O', long = "save-optional", group = "dependency_group")]
     save_optional: bool,
+    /// Install the specified packages as optionalDependencies.
```
Comment (anonrig) [inline crates/cli/src/commands/add.rs:26]: This comment is wrong. 

Using --save-peer will add one or more packages to peerDependencies and install them as dev dependencies.

File: crates/cli/src/commands/add.rs [hunk 2]
```diff
@@ -117,6 +122,13 @@ impl PackageManager {
             &latest_version.serialize(args.save_exact),
             args.dependency_group(),
         )?;
+        if args.dependency_group() == DependencyGroup::Peer {
```
Comment (anonrig) [inline crates/cli/src/commands/add.rs:125]: Can you add a comment in here also similar to "Using --save-peer will add one or more packages to peerDependencies and install them as dev dependencies."

File: crates/cli/src/commands/add.rs [hunk 3]
```diff
@@ -39,6 +42,8 @@ impl AddCommandArgs {
             DependencyGroup::Dev
```
Comment (anonrig) [inline crates/cli/src/commands/add.rs:42]: can you also update crates/cli/readme.md file?

**Comment by vramana** [comment]
@anonrig Fixed all comments

**Review by anonrig — APPROVED** [review]

**Comment by anonrig** [comment]
Thank you!

**Pull request merged**